scipy/linalg/_special_matrices.py

```diff
@@ -443,22 +443,21 @@ def companion(a):
 
     Parameters
     ----------
-    a : (N,) array_like
+    a : (..., N) array_like
         1-D array of polynomial coefficients. The length of `a` must be
         at least two, and ``a[0]`` must not be zero.
```

Comment:
Hmm the documentation here doesn't reflect the new behavior precisely. It could be stated relatively simply like:
```suggestion
        Each slice along the last axis `ai` is a
        1-D array of polynomial coefficients. The length of `ai` must be
        at least two, and ``ai[0]`` must not be zero.
```

or it could be a bit more like code:
```suggestion
        Each slice along the last axis of `a` is a
        1-D array of polynomial coefficients. The length of `a` along the last axis must be
        at least two, and all of `a[..., 0]` must be nonzero.
```